src/main/java/school/lemon/changerequest/java/matrix/Matrix.java

```diff
@@ -0,0 +1,104 @@
+package school.lemon.changerequest.java.matrix;
+
+/**
+ * Created by Yaroslav Pavlinskiy on 12.12.2016.
+ */
+public class Matrix {
+
+    private final int Rows; // count Rows
+    private final int Columns; // count Columns
+    private final int[][] data; // 2d array for matrix
+
+    public Matrix(int R,int C)// Constructor with Rows and Columns
+    {
+        this.Rows = R;
+        this.Columns = C;
+        this.data = new int[R][C];
+    }
+
+    public Matrix(int[][]data) //Constructor 2d array
+    {
+        this.Rows = data.length;
+        this.Columns = data[0].length;
+        this.data = data;
+    }
+    private Matrix(Matrix A) { this(A.data); }
+
+    public Matrix AddMatrix(Matrix B) // Method that adding matrix A and B
+    {
+        return Calculate(B,'+');
+    }
+
+    public Matrix SubtractMatrix(Matrix B)
+    {
+        return Calculate(B,'-');
+    }
+
+    public Matrix MultiplyNumber(int n)
+    {
+        Matrix A = this;
+        Matrix C = new Matrix(Rows, Columns);
+        for (int i = 0; i < Rows; i++)
+            for (int j = 0; j < Columns; j++)
+                C.data[i][j] = A.data[i][j] * n;
+        return C;
+    }
+
+    public  Matrix MultiplyMatrix(Matrix B)
+    {
+        return Calculate(B,'*');
+    }
+
+    public Matrix Transpose()
+    {
+        Matrix A = this;
```

Comment:
Not sure you need extract this to variable. As for me it's even better to use data directly. Like `C.data[i][j] = data[j][i];`